### PR TITLE
apply IAM role to AWS S3 account

### PR DESCRIPTION
**Pull request by Xiaochao8**
Other than applying IAM role to Aws account, this PR upgrades Aws S3 Golang interfaces from V1 to V2.

### CODE REVIEW

File: modules/translation/dao/s3bundle/s3client.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		RoleCredentials: roleOutput.Credentials,
+	}
+
+	writeClient(newInst)
+
+	return newInst
+}
+
+func readClient() RoleClient {
+	clientInstLock.RLock()
```
Comment (tigershi) [inline modules/translation/dao/s3bundle/s3client.go:88]: why need add a lock in read

Comment (Xiaochao8) [reply modules/translation/dao/s3bundle/s3client.go:88]: Here is the RWLock - **clientInstLock** - to protect read and write.

Comment (Xiaochao8) [reply modules/translation/dao/s3bundle/s3client.go:88]: It's possible that multiple threads access the instance symultaneously.

**Review by jin-jasper — DISMISSED** [review]

File: modules/translation/dao/s3bundle/s3client.go (hunk 1 above)
Comment (tigershi) [reply modules/translation/dao/s3bundle/s3client.go:88]: You have used the clientInstLock in write function and it is sync.RWMutex. And you use the other Lock in read function. The one source use the two locks. It means that read function and write function can concurrent execution

Comment (Xiaochao8) [reply modules/translation/dao/s3bundle/s3client.go:88]: The one resource use the same lock `clientInstLock` which is a  `sync.RWMutex`. When reading the variable, use a read lock; When writting the variable, use write lock.
Are there any problems?

**Review by tigershi — APPROVED** [review]

File: modules/translation/dao/s3bundle/s3client.go (hunk 1 above)
Comment (Xiaochao8) [reply modules/translation/dao/s3bundle/s3client.go:83]: Resovled to use a signal variable.

**Review by tigershi — APPROVED** [review]

**Review by jin-jasper — APPROVED** [review]

**Pull request merged**